### PR TITLE
DM-46614: Add option to filter diaObjects for forced measurement

### PR DESCRIPTION
**Pull request by isullivan**

### CODE REVIEW

File: python/lsst/ap/association/diaForcedSource.py [hunk 1]
```diff
@@ -121,7 +128,12 @@ def run(self,
             difference and direct images at DiaObject locations.
         """
 
-        afw_dia_objects = self._convert_from_pandas(dia_objects)
+        nDiaSources = dia_objects.loc[:, "nDiaSources"]
+        objectTable = dia_objects[nDiaSources > self.config.historyThreshold]
```
Comment (ebellm) [inline python/lsst/ap/association/diaForcedSource.py:132]: You could do these two lines more compactly as 

`objectTable = dia_objects.query(f'nDiaSources > {self.config.historyThreshold}')`

File: python/lsst/ap/association/diaForcedSource.py [hunk 2]
```diff
@@ -121,7 +128,12 @@ def run(self,
...
+        if objectTable.empty:
+            return pd.DataFrame()
```
Comment (ebellm) [inline python/lsst/ap/association/diaForcedSource.py:134]: does this need to be wrapped in `self._convert_from_pandas`?

File: python/lsst/ap/association/diaPipe.py [hunk 3]
```diff
@@ -933,3 +928,29 @@ def mergeCatalogs(self, originalCatalog, newCatalog, catalogName):
...
+        """Update the diaObject table with the new diaSource records.
+
+        Parameters
+        ----------
+        diaObjects : `pandas.DataFrame`
+            Table of new DiaObjects merged with their history.
+            This table will be updated in place.
+        diaSources : `pandas.DataFrame`
+            The combined preloaded and associated diaSource catalog.
+        """
```
Comment (ebellm) [inline python/lsst/ap/association/diaPipe.py:942]: I think you can do all of this more efficiently with something like

```
nDiaSources = diaSources[['diaObjectId','diaSourceId']].groupby('diaObjectId').agg(len)
nDiaSources.name = "nDiaSources"
# I don't remember if diaObjects is indexed or not.  If it is, we could do this:
del diaObjects["nDiaSources"]
diaObjects = diaObjects.join(nDiaSources, how="left")
```

**Review by ebellm — CHANGES_REQUESTED** [review]
A few suggestions.  Happy to rubber-duck the pandas if needed.

File: python/lsst/ap/association/diaForcedSource.py (hunk 2 above)
Comment (isullivan) [reply python/lsst/ap/association/diaForcedSource.py:134]: No, the dataframe is coerced to the sdm_schemas format in diaPipe. I'll add a comment.

**Review by ebellm — APPROVED** [review]

**Pull request merged**